File: onlinejudge/leetcode/1-Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] answer = new int[2];
        
        for (int i = 0; i < nums.length - 1; i += 1) {
            for (int j = i + 1; j < nums.length; j += 1) {
                if (nums[i] + nums[j] == target){
                    answer[0] = i;
                    answer[1] = j;
                }
            }
        }
        return answer;
    }
}

File: onlinejudge/leetcode/1-Two Sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  num_to_index = Hash.new

  nums.each_with_index do |num, i|
      complement = target - num

      if num_to_index.key?(complement)
          return [num_to_index[complement], i]
      end

      num_to_index[num] = i
  end

  nil
end